src/rgw/rgw_redis_driver.h

```diff
@@ -0,0 +1,130 @@
+    }
+    virtual ~RedisDriver()
+    {
+      remove_partition_info(partition_info);
+    }
+
+    /* Entry */
+    virtual bool key_exists(const DoutPrefixProvider* dpp, const std::string& key) override;
+    virtual std::vector<Entry> list_entries(const DoutPrefixProvider* dpp) override;
+    virtual size_t get_num_entries(const DoutPrefixProvider* dpp) override;
+    //int update_local_weight(const DoutPrefixProvider* dpp, std::string key, int localWeight); // may need to exist for base class -Sam
+
+    /* Partition */
+    virtual Partition get_current_partition_info(const DoutPrefixProvider* dpp) override { return partition_info; }
+    virtual uint64_t get_free_space(const DoutPrefixProvider* dpp) override { return free_space; } // how to get this from redis server? -Sam
+    static std::optional<Partition> get_partition_info(const DoutPrefixProvider* dpp, const std::string& name, const std::string& type);
+    static std::vector<Partition> list_partitions(const DoutPrefixProvider* dpp);
+
+    virtual int initialize(CephContext* cct, const DoutPrefixProvider* dpp) override;
+    virtual int put(const DoutPrefixProvider* dpp, const std::string& key, bufferlist& bl, uint64_t len, rgw::sal::Attrs& attrs) override;
+    virtual int get(const DoutPrefixProvider* dpp, const std::string& key, off_t offset, uint64_t len, bufferlist& bl, rgw::sal::Attrs& attrs) override;
+    virtual int append_data(const DoutPrefixProvider* dpp, const::std::string& key, bufferlist& bl_data) override;
+    virtual int delete_data(const DoutPrefixProvider* dpp, const::std::string& key) override;
+    virtual int get_attrs(const DoutPrefixProvider* dpp, const std::string& key, rgw::sal::Attrs& attrs) override;
+    virtual int set_attrs(const DoutPrefixProvider* dpp, const std::string& key, rgw::sal::Attrs& attrs) override;
+    virtual int update_attrs(const DoutPrefixProvider* dpp, const std::string& key, rgw::sal::Attrs& attrs) override;
+    virtual int delete_attrs(const DoutPrefixProvider* dpp, const std::string& key, rgw::sal::Attrs& del_attrs) override;
+    virtual std::string get_attr(const DoutPrefixProvider* dpp, const std::string& key, const std::string& attr_name) override;
+    virtual int set_attr(const DoutPrefixProvider* dpp, const std::string& key, const std::string& attr_name, const std::string& attr_val) override;
+
+    virtual std::unique_ptr<CacheAioRequest> get_cache_aio_request_ptr(const DoutPrefixProvider* dpp) override;
+    virtual rgw::AioResultList get_async(const DoutPrefixProvider* dpp, optional_yield y, rgw::Aio* aio, const std::string& key, off_t ofs, uint64_t len, uint64_t cost, uint64_t id) override;
+
+    struct libaio_handler { // should this be the same as SSDDriver? -Sam
+      rgw::Aio* throttle = nullptr;
+      rgw::AioResult& r;
+
+      // read callback
+      void operator()(boost::system::error_code ec, bufferlist bl) const {
+	r.result = -ec.value();
+	r.data = std::move(bl);
+	throttle->put(r);
```

Comment:
according to [the docs for async_exec()](https://boostorg.github.io/redis/classaedis_1_1basic__connection.html#a4d7f180e64db5d4e07f7c62b3aa931d4), this will need to use the signature `void f(boost::system::error_code, std::size_t)`

you'll need to copy the data from the `boost::redis::response` into `r.data`